src/Core/Router.php

```diff
@@ -2,60 +2,52 @@
 
 namespace TrendingRepos\Core;
 
+use TrendingRepos\Exception\RouterException;
+use TrendingRepos\Core\Request;
+
 class Router {
-    public $routes = [
-        'GET' => [],
-        'POST' => []
-    ];
 
-    public static function load($file) {
-        $router = new static;
-        require $file;
+    public $routes;
+    public $request;
 
-        return $router;
+    public function __construct(array $routes, Request $request)
+    {
+        $this->routes = $routes;
+        $this->request = $request;
     }
 
-    public function get($uri, $controller) {
-        $parameters = (new Request)->parameters();
-        if(!empty($parameters)) {
-            foreach($parameters as $key => $value) {
-                $uri = trim($uri, $key . '=' . $value);
-            }
-            $uri = trim($uri, '/?');
+    public function get() {
+        if(array_key_exists($this->request->methodType(), $this->routes)) {
+            $slug = $this->request->getSlug();
 
-           if($uri  == null) {
-               $uri = '';
-           }
+            if($this->uriExists($slug)){
+                $this->direct($this->request->getSlug(), $this->request->methodType());
+            }else {
+                throw new RouterException('This is not the web page you are looking for!');
+            }
+        }else {
+            throw new RouterException('Routes file structure in not as we expect');
         }
-
-        $this->routes['GET'][$uri] = $controller;
     }
 
-    public function post($uri, $controller) {
-        $this->routes['POST'][$uri] = $controller;
+    private function direct(string $uri, string $methodType) {
+        $this->callAction(
+            ...explode('@', $this->routes[$methodType][$uri])
+        );
     }
 
-    public function direct($uri, $methodType) {
-
-        if(array_key_exists($uri, $this->routes[$methodType])) {
-            $some = explode('@',$this->routes[$methodType][$uri]);
-
-            return $this->callAction(
-                ...explode('@', $this->routes[$methodType][$uri])
-            );
-//            return $this->routes[$methodType][$uri];
-        }
-        return view('404');
-    }
-
-    public function callAction($controller, $action){
+    private function callAction(string $controller, string $action){
         $controller = "TrendingRepos\\Controller\\{$controller}";
         $controller = new $controller;
 
-        if(! method_exists($controller, $action)){
-            throw new \Exception("This Action doesn't exit on the controller");
+        if(!method_exists($controller, $action)) {
+            throw new RouterException('This action doesn\' exist!');
         }
 
-        return (new $controller)->$action();
+        (new $controller)->$action();
+    }
+
+    private function uriExists($slug) : bool {
```

Comment:
Please please don't forget type hints. Also, watch the codestyle,  space before `:` is not needed, but after it is:

```suggestion
    private function uriExists(string $slug): bool {
```